python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -728,93 +729,67 @@ def _actually_load_xml(self, fobj, fn, this_id, load_backup):
-                else:
-                    obj.setNodeAttribute(self.sub_split, None)
-
-                for node_key, node_val in obj.getNodeData().iteritems():
-                    if isType(node_val, Node):
-                        if node_key not in Node._ref_list:
-                            node_val._setParent(obj)
-
-                # Check if index cache; if loaded; was valid:
-                if obj.getNodeIndexCache() not in [None, {}]:
-                    new_idx_cache = self.registry.getIndexCache(stripProxy(obj))
-                    if new_idx_cache != obj.getNodeIndexCache():
-                        logger.debug("NEW: %s" % str(new_idx_cache))
-                        logger.debug("OLD: %s" % str(obj.getNodeIndexCache()))
-                        # index is wrong! Try to get read access - then we can fix this
-                        if len(self.lock([this_id])) != 0:
-                            self.index_write(this_id)
-                            # self.unlock([this_id])
-
-                            old_idx_subset = all((k in new_idx_cache and new_idx_cache[k] == v) for k, v in obj.getNodeIndexCache().iteritems())
-                            if not old_idx_subset:
-                                # Old index cache isn't subset of new index cache
-                                new_idx_subset = all((k in obj.getNodeIndexCache() and obj.getNodeIndexCache()[k] == v) for k, v in new_idx_cache.iteritems())
-                            else:
-                                # Old index cache is subset of new index cache so no need to check
-                                new_idx_subset = True
+            has_children = (self.sub_split is not None) and (self.sub_split in obj.getNodeData()) and len(obj.getNodeAttribute(self.sub_split)) == 0
 
-                            if not old_idx_subset and not new_idx_subset:
-                                logger.warning("Incorrect index cache of '%s' object #%s was corrected!" % (self.registry.name, this_id))
-                                logger.debug("old cache: %s\t\tnew cache: %s" % (str(obj.getNodeIndexCache()), str(new_idx_cache)))
-                                self.unlock([this_id])
-                        else:
-                            pass
-                            # if we cannot lock this, the inconsistency is
-                            # most likely the result of another ganga
-                            # process modifying the repo
-                            #obj.setNodeIndexCache(None)
-                #obj.setNodeIndexCache(None)
+            if has_children:
+                obj.setNodeAttribute(self.sub_split, SubJobXMLList.SubJobXMLList(os.path.dirname(fn), self.registry, self.dataFileName, load_backup))
+            else:
+                obj.setNodeAttribute(self.sub_split, None)
+
+            # Check if index cache; if loaded; was valid:
+            if obj.getNodeIndexCache() not in [None, {}]:
```

Comment:
It used to be incorrectly set to none, i don't know if this has been fixed yet. i'll fix this in a future pr once i've done some work on the index cache